File: src/main/java/at/hannibal2/skyhanni/utils/jsonobjects/FriendsJson.java
package at.hannibal2.skyhanni.utils.jsonobjects;

import com.google.gson.annotations.Expose;

import java.util.Map;
import java.util.UUID;

public class FriendsJson {
    @Expose
    public Map<UUID, PlayerFriends> players;

    public static class PlayerFriends {

        @Expose
        public Map<UUID, Friend> friends;

        public static class Friend {
            @Expose
            public String name;
            @Expose
            public boolean bestFriend;
        }
    }
}

File: src/main/java/at/hannibal2/skyhanni/data/jsonobjects/local/FriendsJson.kt
package at.hannibal2.skyhanni.data.jsonobjects.local

import com.google.gson.annotations.Expose
import java.util.UUID


data class FriendsJson(
    @Expose
    val players: MutableMap<UUID, PlayerFriends> = mutableMapOf(),
)

data class PlayerFriends(
    @Expose
    val friends: MutableMap<UUID, Friend> = mutableMapOf(),
)

data class Friend(
    @Expose
    val name: String = "",

    @Expose
    var bestFriend: Boolean = false,
)
